src/icp_canister_backend/tests/test.rs

```diff
@@ -1424,3 +1424,42 @@ fn test_reward_rate_increase_decrease_during_episodes() {
         decreased_reward_rate
     );
 }
+
+#[test]
+fn test_reward_distribution_middle_and_final() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
+    let deposit_amount = Nat::from(100_000_000u64);
+    let reward_amount = Nat::from(50_000_000u64);
+
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 0);
+    create_deposit(&pic, canister_id, ledger_id, user, deposit_amount.clone(), stakable_episode);
+    reward_pool(&pic, canister_id, ledger_id, user, reward_amount.clone())
+        .expect("Reward pool should succeed");
+
+    let earning_duration = get_episode_time_to_end(&pic, stakable_episode);
+
+    //Middle  rewards distribution
+    advance_time(&pic, earning_duration / 2);
+    
+    let middle_rewards = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get middle rewards");
+    
+    let expected_middle = Nat::from(earning_duration / 2);
+    let allowed_error_middle = expected_middle.clone() / Nat::from(100u64);
+    assert_with_error(&middle_rewards, &expected_middle, &allowed_error_middle, "Middle reward distribution");
```

Comment:
Why do you compare time with tokens?
expected_middle is a time variable
and middle_rewards is denominated in token